hack/istio/install-istio-via-sail.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+set -e
+
+if ! command -v yq &> /dev/null; then
+  echo "yq is required but it is either not installed or not in the PATH."
+  exit 1
+fi
+
+CUSTOM_INSTALL_SETTINGS=""
+PATCH_FILE=""
+
+# process command line args
+while [[ $# -gt 0 ]]; do
+  key="$1"
+  case $key in
+    -pf|--patch-file)
+      PATCH_FILE="$2"
+      shift;shift
+      ;;
+    -s|--set)
+      if [ -n "${CUSTOM_INSTALL_SETTINGS}" ]; then
+        CUSTOM_INSTALL_SETTINGS="${CUSTOM_INSTALL_SETTINGS} | $2"
+      else
+        CUSTOM_INSTALL_SETTINGS="$2"
+      fi
+      shift;shift
+      ;;
+    -h|--help)
+      cat <<HELPMSG
+Valid command line arguments:
+  -pf|--patch-file <name=value>:
+       filepath to a yaml file of an 'Istio' resource that will overlay the default 'Istio' resource. 
+       --patch-file /path/to/patch-file.yaml
+  -s|--set <name=value>:
+       Sets a name/value pair for a custom install setting. Some examples you may want to use:
+       --set installPackagePath=/git/clone/istio.io/installer
+  -h|--help:
+       this message
+HELPMSG
+      exit 1
+      ;;
+    *)
+      echo "ERROR: Unknown argument [$key]. Aborting."
+      exit 1
+      ;;
+  esac
+done
+
+helm upgrade sail-operator sail-operator \
+  --install \
+  --create-namespace \
+  --namespace sail-operator \
+  --repo https://istio-ecosystem.github.io/sail-operator
```

Comment:
This assumes the sail-operator repo is already added to helm, which likely isn't going to be the case (almost certainly for the first time someone runs this). I would also update the helm repo just as a convienence, in order to pick up the latest updates for the user if they already added the repo earlier:

So above this `helm upgrade` I would do this:

```bash
if ! helm repo list | grep -q "sail-operator"; then
    echo "Adding Helm repo 'sail-operator'"
    echo helm repo add "sail-operator" "https://istio-ecosystem.github.io/sail-operator"
else
    echo helm repo update "sail-operator"
fi
```

Reply:
The `--repo https://istio-ecosystem.github.io/sail-operator` flag will use the url instead of looking at your local charts repo.

Reply:
Interesting. So `--repo` will pull the latest charts. TIL...